### PR TITLE
Add details of set_pull method and default pull states.

### LINKED ISSUE #381: Docs do not list the set_pull methods
The docs here:

https://microbit-micropython.readthedocs.io/en/latest/pin.html#pin-functions

do not list the set_pull() method.

From python-microbit mailing list...

Hi David,

Looking at the code the default pull states seem to be:

read_analog: configured by mbed analogin_init() function, looks like
there is no pull
read_digital: PULL_DOWN
is_touched: PULL_UP

To change the value, use: pin.set_pull(pin.PULL_UP), or PULL_DOWN or NO_PULL.

### PR DESCRIPTION
**Pull request by ntoll**
Fixes #381

### CODE REVIEW

File: docs/pin.rst [hunk 1]
```diff
@@ -170,3 +176,10 @@ to that.
         whatever is connected to it. Human body has quite a large capacitance,
         so touching the pin gives a dramatic change in reading, which can be
         detected.
+
+
+The default pullup/down states for each type of pin are as follows:
+
+* ``read_analog`` - there is no pull.
+* ``read_digital`` - ``pin.PULL_DOWN``
+* ``is_touched`` - ``pin.PULL_UP``
```
Comment (dpgeorge) [inline docs/pin.rst]: It's a bit confusing listing it like this.  First, read_analog/read_digital/is_touched are not pins but rather states of a pin (ie a particular mode).  Second, you can't actually change the pull for any mode except read_digital.

So I'd suggest some wording like this: "The pull mode for a pin is automatically configured when the pin changes to an input mode.  Input modes are when you call read_analog/read_digital/is_touched.  The pull mode for these is, respectively, NO_PULL, PULL_DOWN, PULL_UP.  Only when in read_analog mode can you call set_pull to change the pull mode from the default."

**Comment by ntoll** [comment]
Fixed and rebased.

**Comment by whaleygeek** [comment]
Also note that the micro:bit has external weak (10M) pull-ups fitted on P0 P1 and P2 only, in order for the touch sensing to work. See the edge connector data sheet here: http://tech.microbit.org/hardware/edgeconnector_ds/

File: docs/pin.rst [hunk 2]
```diff
@@ -170,3 +176,9 @@ to that.
         whatever is connected to it. Human body has quite a large capacitance,
         so touching the pin gives a dramatic change in reading, which can be
         detected.
+
+The pull mode for a pin is automatically configured when the pin changes to an
+input mode. Input modes are when you call ``read_analog`` / ``read_digital`` /
+``is_touched``. The pull mode for these is, respectively, ``NO_PULL``,
+``PULL_DOWN``, ``PULL_UP``. Only when in ``read_analog`` mode can you call
```
Comment (dpgeorge) [inline docs/pin.rst]: Sorry, my bad, read_analog should be read_digital on this line.

**Comment by ntoll** [comment]
I think it 's probably worth adding @whaleygeek's comment somewhere in this documentation too.

**Comment by dpgeorge** [comment]
Thanks @ntoll!  Good that you added the link to the edge-connector details.

**Pull request merged**